### ISSUE TITLE
Investigate duration query concern

### ISSUE THREAD

**Issue by codefromthecrypt** [openzipkin/zipkin#2289]
@drolando I forget if you are using cassandra or not. It will help decide if the duration query by all will never work or if it is just asking for too many service names. Can you mention which storage you are using and if you select anything else when doing duration, also how many service names do you have?

>  Filtering by duration with "service: all" results in a timeout every time. If this will never work, should we have a JS check that warns against it? (TBH I think the current UI also has this same issue)

**Comment by drolando** [comment]
Yeah, I'm using Cassandra 3.11.3 with the new schema. ~200 service names in dev.

I didn't select anything at all. I just opened the search page, left everything to their default and just inserted "100ms" as duration filter. (Same issue happen whatever I write in that textbox, so it's not an issue woth @tacigar's PR).
The errors are all timeouts while reading from Cassandra.

**Comment by codefromthecrypt** [comment]
thanks for the context

**Comment by codefromthecrypt** [comment]
it is possible we could allow a setting to disable the "all" search control as indeed it is the most expensive query. meanwhile there could be a way to improve the performance (like by redoing the query in some way that isn't N+1 in nature) cc @openzipkin/cassandra
